### DISCUSSION TITLE
Can I use remix-auth without session storage?

### DISCUSSION THREAD

**Discussion by alvinthen** [sergiodxa/remix-auth#230] [Q&A]
Hi, I’m using remix to build an API server for mobile app usage. I want to use remix-auth for authentication. I supposed I will be using openID or jwt strategy. Can I use it without session storage, since my APIs should be restful. Thanks

**Comment by sergiodxa** [comment]
The sessionStorage is always required, but if you don't set `successRedirect` then `authenticate` will give you back the user data from the strategy instead of saving it to the session, you can then use that to send to your mobile app what you need.

But I think Remix Auth will not give you too many benefits in an API case, the OAuth flows will be implemented in the mobile app and you can pass an accessToken from the OAuth provider to your API to complete the login, for email/user+password it's not hard to send them in a POST request.
